### PR TITLE
Query and update animals table

### PR DESCRIPTION
**Pull request by anderCM**
- Add a column species of type string to your animals table. Modify your schema.sql file.
- Insert the following data:
  - Animal: His name is Charmander. He was born on Feb 8th, 2020, and currently weighs -11kg. He is not neutered and he has never tried to escape.
  - Animal: Her name is Plantmon. She was born on Nov 15th, 2021, and currently weighs -5.7kg. She is neutered and she has tried to escape 2 times.
  - Animal: His name is Squirtle. He was born on Apr 2nd, 1993, and currently weighs -12.13kg. He was not neutered and he has tried to escape 3 times.
  - Animal: His name is Angemon. He was born on Jun 12th, 2005, and currently weighs -45kg. He is neutered and he has tried to escape once.
  - Animal: His name is Boarmon. He was born on Jun 7th, 2005, and currently weighs 20.4kg. He is neutered and he has tried to escape 7 times.
  - Animal: Her name is Blossom. She was born on Oct 13th, 1998, and currently weighs 17kg. She is neutered and she has tried to escape 3 times.
  - Animal: His name is Ditto. He was born on May 14th, 2022, and currently weighs 22kg. He is neutered and he has tried to escape 4 times.
- Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.
- Inside a transaction:
  - Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
  - Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
  - Commit the transaction.
  - Verify that change was made and persists after commit.
- Take a screenshot of the results of your actions.
- Now, take a deep breath and... Inside a transaction delete all records in the animals table, then roll back the transaction.
- After the rollback verify if all records in the animals table still exists. After that, you can start breathing as usual ;)
- Take a screenshot of the results of your actions.
- Inside a transaction:
- Delete all animals born after Jan 1st, 2022.
  - Create a savepoint for the transaction.
  - Update all animals' weight to be their weight multiplied by -1.
  - Rollback to the savepoint
  - Update all animals' weights that are negative to be their weight multiplied by -1.
  - Commit transaction
- Write queries to answer the following questions:
  - How many animals are there?
  - How many animals have never tried to escape?
  - What is the average weight of animals?
  - Who escapes the most, neutered or not neutered animals?
  - What is the minimum and maximum weight of each type of animal?
  - What is the average number of escape attempts per animal type of those born between 1990 and 2000?

## Images
UPDATE POKEMON DIGIMON

transaction rollback delete

update weight transaction

How many animals are there

How many animals have never tried to escape

What is the average weight of animals

Who escapes the most, neutered or not neutered animals

What is the minimum and maximum weight of each type of animal

What is the average number of escape attempts per animal type of those born between 1990 and 2000

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -32,4 +32,74 @@ SELECT * FROM animals
 /* Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg) */
 SELECT * FROM animals
 	WHERE weight_kg >= 10.4
-		AND weight_kg <= 17.3;
\ No newline at end of file
+		AND weight_kg <= 17.3;
```
Comment (Davidosky007) [inline queries.sql:35]: * According to the [requirements](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=Inside%20a%20transaction%20update%20the%20animals%20table%20by%20setting%20the%20species%20column%20to%20unspecified.%20Verify%20that%20change%20was%20made.%20Then%20roll%20back%20the%20change%20and%20verify%20that%20the%20species%20columns%20went%20back%20to%20the%20state%20before%20the%20transaction.) you are to write queries that do the following inside your transactions, Kindly add them here 🙏 

> Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.

File: queries.sql [hunk 2]
```diff
@@ -32,4 +32,74 @@ SELECT * FROM animals
...
+	Commit the transaction. */
+BEGIN;		
+UPDATE animals 
+	SET species = CASE
+					WHEN name LIKE '%mon'
+						THEN 'digimon'
+					ELSE
+						'pokemon'
+					END;
+COMMIT;
```
Comment (Davidosky007) [inline queries.sql:50]: * * According to the [requirements](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=Inside%20a%20transaction%20update%20the%20animals%20table%20by%20setting%20the%20species%20column%20to%20unspecified.%20Verify%20that%20change%20was%20made.%20Then%20roll%20back%20the%20change%20and%20verify%20that%20the%20species%20columns%20went) you are to 

> Verify that the change was made.

> verify that the change persists after commit 
* HINTS you can do all these by following these steps 
* After setting species , then
* You check to see if species were actually set by running `SELECT species from animals;` 
*  Then commit the transaction and verify that the change persists by running ``SELECT species from animals;``

File: data.sql [hunk 3]
```diff
@@ -3,4 +3,14 @@ INSERT INTO
...
\ No newline at end of file
+			(4, 'Devimon', '2017-05-12', 5, true, 11);
+
+INSERT INTO 
+	animals(id, name, date_of_birth, escape_attempts, neutered, weight_kg)
+	VALUES	(5, 'Charmander', '2020-02-08', 0, false, -11),
+			(6, 'Plantmon', '2021-11-15', 2, true, -5.7),
+			(7, 'Squirtle', '1993-04-02', 3, false, -12.13),
+			(8, 'Angemon', '2005-06-12', 1, true, -45),
+			(9, 'Boarmon', '2005-06-05', 7, true, 20.4),
```
Comment (Davidosky007) [inline data.sql]: * Kindly correct the DOB of animal `Boarmon` is should be `born on Jun 7th, 2005` [Check requirements](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=born%20on%20Jun%207th%2C%202005)

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @anderCM 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* Well written PR 👍 
* Good looking readme 👍 
* Proper use of Gitflow 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql [hunk 4]
```diff
@@ -32,4 +32,89 @@ SELECT * FROM animals
...
+
+SELECT species from animals;
+
+
+/* Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made.
+	Then roll back the change and verify that the species columns went back to the state before the transaction */
+BEGIN;
+SAVEPOINT trs_upd_spec_unde;
+UPDATE animals SET species = 'unspecified';
```
Comment (Davidosky007) [inline queries.sql:58]: * Good job working on the requested changes, just some few requirements are left
* Here also you are [required](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=Inside%20a%20transaction%20update%20the%20animals%20table%20by%20setting%20the%20species%20column%20to%20unspecified.%20Verify%20that%20change%20was%20made.%20Then%20roll%20back%20the%20change%20and%20verify%20that%20the%20species%20columns%20went%20back%20to%20the%20state%20before%20the%20transaction.) to confirm the following. you can use the same HINT as before for this too 
* Also Kindly differentiate the transactions for clarity the queries should be arranged properly, and ensure only related transaction queries are written together 
 > Verify that the change was made.

> roll back the change 

> verify that the species columns went back to the state before the transaction.

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @anderCM 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
* Proper use of Gitflow 👍 
* Professional PR 👍 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidosky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TadesseDev — APPROVED** [review]
### Hi @anderCM 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy (1)



## Highlights
 - [x] Good job making the changes requested by the previous code reviewer. ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 1 above)
Comment (anderCM) [reply queries.sql:35]: Thanks for the information. I will update it

File: queries.sql (hunk 2 above)
Comment (anderCM) [reply queries.sql:50]: Thanks for the information. I will update it

File: data.sql (hunk 3 above)
Comment (anderCM) [reply data.sql]: Thanks for the information. I will update it

File: queries.sql (hunk 4 above)
Comment (anderCM) [reply queries.sql:58]: Thanks for the information. I will update it

**Pull request merged**